web/src/lib.rs

```diff
@@ -81,12 +89,17 @@ pub async fn init_server(app_state: AppState) -> Result<()> {
             Method::PUT,
             Method::PATCH,
         ])
+        // Essential to allow credentials through a reverse proxy like nginx
         .allow_credentials(true)
         // Allow and expose the X-Version header across origins
         .allow_headers([
             ApiVersion::field_name().parse::<HeaderName>().unwrap(),
             AUTHORIZATION,
             CONTENT_TYPE,
+            // Headers that nginx reverse proxy might forward
```

Comment:
This is awesome! Would you mind also adding `x-request-id`? 

It's insanely useful for observability 
https://http.dev/x-request-id

Reply:
Fabulous suggestion, I wasn't aware of this header!

Reply:
Yes definitely

Reply:
@jhodapp I think it might actually make sense for nginx to to generate the values on each request. It's commonly done at the system edge so that every request coming in includes the header and it's value. The BE then reads the value, adds it to every log statement output for the request,  and adds it back as the `x-request-id` in responses to the FE (or to other services if they exist). That way we can trace the whole round trip for each request via logs and we'll be able to tell which logs belong to which request. It becomes necessary when there are hundreds or thousands of parallel requests being served.

Reply:
With my change, we actually get both. If it's provided, forward it. If it's not, nginx will automatically generate a UUIDv4 unique id.